### DISCUSSION TITLE
First feedback on latest 15.7 build

### DISCUSSION THREAD

**Discussion by spanzetta** [jomjol/AI-on-the-edge-device#2926] [General]
- wrong raw/value detection/+1 m3
appears to be solved (same as 15.4)

- Memory leak
appears to be solved (same as 15.6)

- new camera settings
zoom and image settings has some problems... not consistent (if you same things sometimes you get different results). Zoom in particular has problems.
Image quality is ok (need to play with new settings)

- new camera settings interface 
question: are all  these *) normal?
![Screenshot_20240217_102641_Chrome.jpg](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/9310856d-a4f9-40b4-9d1f-e406b5159322)

**Comment by caco3** [comment]
> new camera settings
zoom and image settings has some problems... not consistent (if you same things sometimes you get different results). Zoom in particular has problems.
Image quality is ok (need to play with new settings)

Thanks for the feedback. Can you detail it a bit more?
@SybexX FYI

> new camera settings interface
question: are all these *) normal?

Yes. It was already like this before (only applied on next image), but now we added the `*)` to make this clearer.

**Comment by spanzetta** [comment]
> Thanks for the feedback. Can you detail it a bit more?

It's quite complicated because I cannot reproduce same behaviour all the times..
In few words I noticed that:
- when I enable Zoom and then Update image.. most of the time zoom is not applied..
Then you change another parameters, do update again and the zoom is applied 
- when Zoom is applied when I try to change Y value (to shift vertically) and then update image I see that image is not moved at all.. then I change X value, do update image and eventually the Y shift is applied.. 
- some parameters have big impact on image quality .. for example the sharpness.. change by one step and image quality changes a lot..
I hope it will improved in the future..  the interface now is very complete!

Thanks a lot to all developers for this great sw!!

**Reply by spanzetta** [reply]
another one:
I play with zoom in the reference image settings.. I exit without saving but the detection now works on the zoomed image (even if I didn't save the new reference image with zoom)
After a forced reboot it returns working normally

**Comment by Piti61** [comment]
After installing 15.7, the images are significantly darker and cannot be brightened much via "LED intensity" (I use two external LEDs). Back to 15.6 and the images are nice and bright again. The whole thing can be reproduced.

**Reply by spanzetta** [reply]
If you enable expert settings and the Auto Exposure control you will get the right brightness.. and now you have much more parameters to fine tune your image..
Try it..

**Comment by friedpa** [comment]
+1 at my environment

**Comment by SybexX** [comment]
The problem with the zoom is due to the camera module, the zoom function is actually not a real zoom^^
With the zoom function, the camera only sends the defined area and not the entire image.
And there are probably problems when sending the defined area, these can be:
- the image is transferred incompletely (image error)
- the desired image will be sent as a second image, i.e. if you press "update Reference Image" a few times
- and only works reasonably reliably in 4:3 format...
Therefore, I would only use this function if manual focusing is not possible or is not sufficient.
And there are certain restrictions on the other functions, i.e. if you switch on a certain function, others are not available or are overwritten.
These problems can probably only be solved by removing the functions again.

**Comment by SybexX** [comment]
I found the following statement:

You have to be very careful what combination of parameters you use and in what order you use them.
I had more success by commenting out the parameters not needed.
Extra care has to be taken when any of the Auto controls are used, as subsequent parameters are ignored.

https://randomnerdtutorials.com/esp32-cam-ov2640-camera-settings/

**Comment by SusiKuller** [comment]
Hi :) Many thanks for the update.

I just installed them on two of my three devices (coming from 15.4). One uses two external LEDs which are now very dark, as described above.  When I want to update the reference image ("create new...") to adjust the exposure,  it hangs in the image loading screen ("taking new image"). I tried on my PC and on my mobile phone. 

Images are still being taken and analyzed. Often there is the notification "Initialzing (delayed)" but after some time the flow is succesfully finished. 

"Livestream uses" the internal LED instead of the external ones. 

The one with the internal LED works fine.

**Reply by tglabcz** [reply]
Hi,

Same experience here.
In addition, every time when I try to create new reference image my device rebooted and in log I see:
_[0d00h00m05s] 2024-02-29T13:54:08 <WRN> [MAIN] Reset reason: Exception/panic
[0d00h00m05s] 2024-02-29T13:54:08 <WRN> [MAIN] Device was rebooted due to a software exception! Log level is set to DEBUG until the next reboot. Flow init is delayed by 5 minutes to check the logs or do an OTA update
[0d00h00m05s] 2024-02-29T13:54:08 <WRN> [MAIN] Keep device running until crash occurs again and check logs after device is up again_

**Reply by tglabcz** [reply]
Hi again,
Just FYI, I deleted config.ini file and I did upload of old one from backup, after reboot it seems to work.
If I compare config.ini files only difference is in section [TakeImage], those settings are only in new config.ini file:
;Sharpness = 
;Zoom = false
;ZoomMode = 0
;ZoomOffsetX = 
;ZoomOffsetY = 
;Grayscale = false
;Negative = false
;Aec2 = false
;AutoExposureLevel = 

All are disabled so do not understand why causing issues.

**Reply by tglabcz** [reply]
Hi again,
Seems I found solution, if I enable all options and set them to default values issue is solved:
Sharpness = 0
Zoom = false
ZoomMode = 0
ZoomOffsetX = 0
ZoomOffsetY = 0
Grayscale = false
Negative = false
Aec2 = false
AutoExposureLevel = 0

**Comment by SybexX** [comment]
The parameters are sent to the cam and when they are commented out there are permanent restarts.
You should only edit config.ini manually if you know what you are doing.

**Reply by tglabcz** [reply]
Just for clarification.
Parameters has been commented out by default (after OTA update) not by me.
When I manually enabled those, issue with restart has been solved.

**Comment by SusiKuller** [comment]
My two config.ini do look different. 

Working (internal LED)
![Screenshot_20240301-233321~2](https://github.com/jomjol/AI-on-the-edge-device/assets/155202683/c5a77914-1a00-4c61-95a7-58badf99f342)

Not Working (external LED)
![Screenshot_20240301-233245~2](https://github.com/jomjol/AI-on-the-edge-device/assets/155202683/4a61b2ef-44c6-4995-a2da-2a960eede603)

I also have those commented out entries by default. Will check if viky2379 solution works.

**Comment by SybexX** [comment]
Anyone who wants to can try this firmware.
This is the rolling with a revision of the camera settings.
!!!Important!!! Be sure to use all sd-card files!!!Important!!!
[AI-on-the-edge-device__update__cam_settings_min_(d9de8ac).zip](https://github.com/jomjol/AI-on-the-edge-device/files/14914457/AI-on-the-edge-device__update__cam_settings_min_.d9de8ac.zip)

**Comment by friedpa** [comment]
@SybexX: what to you mean with the last sentence and is this version for OTA?

**Comment by SybexX** [comment]
Since implementing the camera settings only caused problems, I reworked everything.
To avoid unwanted problems, all files from the sd-card folder should be used.
Now I don't know what exactly you mean by "is this version for OTA?" mean. 
I suspect whether you can update the SD card files and the firmware via OTA? 
So far I have never updated the SD card files with OTA, only manually with the "Fileserver" page or with the PC. 
I always delete the existing files and then copy the new ones onto them, then I do a firmware update via the "OTA Update" page. 
If I don't delete the files to be replaced beforehand, I get an error saying the file wasn't replaced or something.

The following files have been changed and must be copied to the SD:

sd-card/html/edit_config.html
sd-card/html/edit_reference.html
sd-card/html/index.html
sd-card/html/overview.html
sd-card/html/readconfigparam.js
sd-card/html/version.txt

sd-card/config/config.ini

the rest remained the same

for OTA Update:
[AI-on-the-edge-device__manual-setup__cam_settings_(d9de8ac).zip](https://github.com/jomjol/AI-on-the-edge-device/files/14914478/AI-on-the-edge-device__manual-setup__cam_settings_.d9de8ac.zip)
[AI-on-the-edge-device__remote-setup__cam_settings_(d9de8ac).zip](https://github.com/jomjol/AI-on-the-edge-device/files/14914484/AI-on-the-edge-device__remote-setup__cam_settings_.d9de8ac.zip)
[AI-on-the-edge-device__update__cam_settings_(d9de8ac).zip](https://github.com/jomjol/AI-on-the-edge-device/files/14914487/AI-on-the-edge-device__update__cam_settings_.d9de8ac.zip)
[AI-on-the-edge-device__update__cam_settings_min_(d9de8ac).zip](https://github.com/jomjol/AI-on-the-edge-device/files/14914490/AI-on-the-edge-device__update__cam_settings_min_.d9de8ac.zip)

![cam_zoom](https://github.com/jomjol/AI-on-the-edge-device/assets/587201/8d4b0237-2d2c-4a6b-bd45-9e6681ca43a8)

**Reply by MVerBerkmoes** [reply]
Confirming this update works.  All file deleted/updated as per the "Fileserver" page suggested above.  Firmware them updated OTA with AI-on-the-edge-device__update__cam_settings_.44d476d.zip then rebooted.

MUCH better response of the webpage and image creations.  No errors in the log, all seems to be working as expected.

Thank you for the effort on this project!

**Reply by MedicusOne** [reply]
Confirm, this update runs over a week without problems.

**Comment by friedpa** [comment]
Running smoothly! Thanks!
